.github/workflows/code-quality.yml

```diff
@@ -98,4 +98,4 @@ jobs:
           --preset sanitize
 
       - name: Build and test
-        run: ctest -VV --test-dir build
+        run: ctest -VV --test-dir build --parallel 2
```

Comment:
Would it not be better to determine the ideal process count from the cmake, like this in [ProcessorCount cmake docs](https://cmake.org/cmake/help/latest/module/ProcessorCount.html)

```
include(ProcessorCount)
ProcessorCount(N)
if(NOT N EQUAL 0)
  set(CTEST_BUILD_FLAGS -j${N})
  set(ctest_test_args ${ctest_test_args} PARALLEL_LEVEL ${N})
endif()
```

Reply:
It would be great if the process count could be automatic.  How could we pass along the `ctest_test_args` variable to ctest, though?  I'll do some doc hunting.

Reply:
Ah, the docs you linked use `ctest -S`.  We could add such a script as a convenience, but the ultimate would be to simply run `ctest` and have the `--parallel` determined automatically.